[vis/resizeChecker] swap out implemenation with ResizeObserver polyfill

spalger opened the pull request:
Discovered the [ResizeObserver](http://rawgit.com/WICG/ResizeObserver/master/index.html) spec and found a polyfill for it that is far faster than the polling done by the previous implementation

spalger commented:
jenkins, test this

ppisljar commented:
tests are failing, but it looks its not related to this PR ( broken master ? ):
```
FAIL: chrome on any platform - kibana - console app - console app - should show the default request (40675ms) expected
'GET _search\n  {\n    "query": { "match_all": {} }\n  }' to equal
'GET _search\n{\n  "query": {\n    "match_all": {}\n  }\n}'
```

ppisljar commented:
functionallity wise everything seems to work fine

spalger commented:
> tests are failing, but it looks its not related to this PR 

The console actually also uses the resize checker, so I moved it out of the vislib and updated the resize checker to support jQuery elements again to fix compatibility.

ppisljar commented:
jenkins, test this

ppisljar commented:
what do you think about moving this under ui/vis/components making it clear that its a component visualizations (even 3rd party) should use?

spalger commented:
I don't feel strongly either way but the resize checker:

 - exports a provider so it would be:

    ```js
    import {Tooltip, ResizeCheckerProvider, Colors} from 'ui/vis/components'
    ```

 - is not always used by visualizations, so console would need to import from `ui/vis/components` as well
 - is not a visual component, which seems to conflict with the definition of "components" I'm most familiar with.

Bargs commented on src/ui/public/resize_checker/resize_checker.js:
```diff
@@ -0,0 +1,94 @@
+  }
+
+  function getSize(el) {
+    return [el.clientWidth, el.clientHeight];
```
Could this trigger costly layout calculations? Why not use the values provided to the callback? http://rawgit.com/WICG/ResizeObserver/master/index.html#resizeobserverentry

spalger replied:
I'm not sure, can it? I went this route because we need to calculate the size in `modifySizeWithoutTriggeringResize()` and I didn't want to try to replicate the behavior necessary to calculate the width/height provided by the clientRect (see https://github.com/que-etc/resize-observer-polyfill/issues/4)

spalger replied:
Also, it's only called when we are trying to ignore a resize event (on load and in `modifySizeWithoutTriggeringResize()`) so I don't think it will be too costly.

Bargs replied:
Ah, I was thinking it ran on every resize event. Maybe it's not too bad then. It looks like `clientWidth` and `clientHeight` [do force layout](https://gist.github.com/paulirish/5d52fb081b3570c81e3a). Not sure how else to implement this resize ignoring method though. Maybe it would be worth adding a comment on `modifySizeWithoutTriggeringResize` saying "This method can trigger forced layouts, use with caution"?

I think this is a great PR btw! The forced layout thing just jumped out at me because I've been dealing with the same problem elsewhere in the code.

spalger replied:
👍 sounds good

Stacey-Gammon commented:
Any reason this wasn't backported?

spalger commented:
Perhaps just the size and lack of need, is there a reason we should?
